src/components/TradeWidget/index.tsx

```diff
@@ -1,25 +1,32 @@
 import React, { useState, useMemo } from 'react'
+import { faExchangeAlt, faPaperPlane } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import styled from 'styled-components'
+import useForm, { FormContext } from 'react-hook-form'
 
-import Widget from 'components/layout/Widget'
-import { useWalletConnection } from 'hooks/useWalletConnection'
 import { Network, TokenDetails } from 'types'
 import { tokenListApi } from 'api'
+import { useTokenBalances } from 'hooks/useTokenBalances'
+import { useWalletConnection } from 'hooks/useWalletConnection'
 import TokenRow from './TokenRow'
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faExchangeAlt, faPaperPlane } from '@fortawesome/free-solid-svg-icons'
+import Widget from 'components/layout/Widget'
 
-function _getToken(symbol: string, tokens: TokenDetails[]): TokenDetails {
-  return tokens.find(token => token.symbol == symbol)
+function _getToken<T extends TokenDetails>(symbol: string, tokens: T[]): T {
```

Comment:
same as above - may be better to use default param for `tokens`

Reply:
Addressed on https://github.com/gnosis/dex-react/pull/163